### ISSUE TITLE
Add accessibility to CheckableImageButton

### ISSUE THREAD

**Issue by Sdaswani** [mozilla-mobile/firefox-tv#528]
This already has one commit, which will make this feature barely accessible, but still needs:
- [x] Announce checked state on change
- [x] Announce checked state on focus (VoiceView handles order differently from TalkBack/developer doc recommendations)

This will involve either experimenting with the docs more for Amazon's VoiceView (see [wiki page](https://github.com/mozilla-mobile/firefox-tv/wiki/Accessibility-Caveats) for very basic docs) or figuring out some other way how Amazon is calling through the accessibility functions.

**Comment by lime124** [comment]
this is for the accessibility comment from eitan? i don't think he meant a literal checkbox. we discussed over slack and he didn't mention needing to change the UI. @Sdaswani

**Comment by Sdaswani** [comment]
cool @lime124 @eeejay so is there any work to do here?

**Comment by mcomella** [comment]
The CheckBox UI widget gives us these a11y wins automatically but afaik, there is some restriction that prevented us from using it – @liuche? 

We could potentially try to extract the a11y code from the CheckBox widget, so getting it's a11y functionality without using the widget directly

**Comment by liuche** [comment]
Shouldn't be hard, I thought we got those for free with my implementation but I think we just need to override some of these functions: https://developer.android.com/guide/topics/ui/accessibility/custom-views.html#populate-events

Bonus is that we could add our own custom accessibility text for turbo mode/pinning if we'd like.

**Comment by eeejay** [comment]
Yeah, didn't look at the code, but if you have your own subclassed button
its easy to override the right methods. I would override
onInitializeAccessibilityNodeInfo() as well.

**Comment by mcomella** [comment]
Best to do 2.0 but if it drops to 2.1, that might be okay too.

**Comment by liuche** [comment]
Pushed a PR to this, which will make this feature barely accessible, but still needs:
- [ ] Announce checked state on focus

This will involve either experimenting with the docs more for Amazon's VoiceView (in [wiki page](https://github.com/mozilla-mobile/firefox-tv/wiki/Accessibility-Caveats) or decompiling the VoiceView code to see what order it calls things in.

**Comment by liuche** [comment]
Unassigning now that first part is landed.
